crates/primitives/src/bits/fixed.rs

```diff
@@ -251,21 +255,44 @@ impl<const N: usize> fmt::UpperHex for FixedBytes<N> {
     }
 }
 
+impl<const N: usize> ops::BitAndAssign<&Self> for FixedBytes<N> {
+    #[inline]
+    fn bitand_assign(&mut self, rhs: &Self) {
+        // Note: `slice::Iter`` has better codegen than `array::IntoIter``
+        iter::zip(self, rhs).for_each(|(a, b)| *a &= *b);
+    }
+}
+
+impl<const N: usize> ops::BitOrAssign<&Self> for FixedBytes<N> {
+    #[inline]
+    fn bitor_assign(&mut self, rhs: &Self) {
+        // Note: `slice::Iter` has better codegen than `array::IntoIter`
+        iter::zip(self, rhs).for_each(|(a, b)| *a |= *b);
+    }
+}
+
+impl<const N: usize> ops::BitXorAssign<&Self> for FixedBytes<N> {
+    #[inline]
+    fn bitxor_assign(&mut self, rhs: &Self) {
+        // Note: `slice::Iter` has better codegen than `array::IntoIter`
+        iter::zip(self, rhs).for_each(|(a, b)| *a ^= *b);
+    }
+}
```

Comment:
these look good, can we group them to the existing `BitX` implements so that we have all of them together.

the old impls of those traits can now just call `self.bitxor_assign(&rhs)` this way we can reuse some code